transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/AbstractHttpJob.java

```diff
@@ -189,6 +189,12 @@ private void writeMetadata( final Transfer target, final ObjectMapper mapper )
             return;
         }
 
+        if ( target.getPath().endsWith( ".md5" ) || target.getPath().endsWith(".sha1") )
```

Comment:
Maybe also sha256, sha512, etc. Galley a AbstractChecksumGeneratorFactory and a few impl classes. Each impl class has a fileExtension. I think maybe we shuld share something instead of hardcoding them here?

Reply:
Maybe create an enum of checksums and save the file ext together. And use them elsewhere.

Reply:
Ah, see. Good to know that we have the impl classes, i will see how to do it later. thanks.